packages/quick_actions/quick_actions_ios/example/ios/RunnerUITests/RunnerUITests.swift

```diff
@@ -73,24 +73,50 @@ class RunnerUITests: XCTestCase {
       )
     }
 
-    quickActionsAppIcon.press(forDuration: 2)
+    findAndTapQuickActionButton(buttonName: "Action one", quickActionsAppIcon: quickActionsAppIcon, springboard: springboard);
 
-    let actionOne = springboard.buttons["Action one"]
-    if !actionOne.waitForExistence(timeout: elementWaitingTime) {
+    let actionOneConfirmation = exampleApp.otherElements["action_one"]
+    if !actionOneConfirmation.waitForExistence(timeout: elementWaitingTime) {
       XCTFail(
-        "Failed due to not able to find the actionOne button from springboard with \(elementWaitingTime) seconds. Springboard debug description: \(springboard.debugDescription)"
+        "Failed due to not able to find the actionOneConfirmation in the app with \(elementWaitingTime) seconds. Springboard debug description: \(springboard.debugDescription)"
       )
     }
 
-    actionOne.tap()
+    XCTAssert(actionOneConfirmation.exists)
+  }
 
-    let actionOneConfirmation = exampleApp.otherElements["action_one"]
-    if !actionOneConfirmation.waitForExistence(timeout: elementWaitingTime) {
+  private func findAndTapQuickActionButton(buttonName: String, quickActionsAppIcon: XCUIElement, springboard: XCUIElement) {
+    var actionButton: XCUIElement?
+    var pressDuration = quickActionPressDuration
+    for _ in 1...quickActionMaxRetries {
+      if !quickActionsAppIcon.waitForExistence(timeout: elementWaitingTime) {
+        XCTFail(
+          "Failed due to not able to find the example app from springboard with \(elementWaitingTime) seconds. Springboard debug description: \(springboard.debugDescription)"
+        )
+      }
+      quickActionsAppIcon.press(forDuration: pressDuration)
+      actionButton = springboard.buttons[buttonName]
+      if actionButton!.waitForExistence(timeout: elementWaitingTime) {
+        // find the button, exit the retry loop.
+        break
+      }
+      let deleteButton = springboard.buttons["DeleteButton"]
+      if deleteButton.waitForExistence(timeout: elementWaitingTime) {
+        // Found delete button instead, we pressed too long, reduce the press time.
+        pressDuration -= pressDurationRetryAdjustment
+      } else {
+        // Neither action button nor delete button was found, we need a longer press.
+        pressDuration += pressDurationRetryAdjustment
```

Comment:
This makes sense to me. though the drawback is that it makes the test a bit complicated, as test code itself is not tested. i would simply try like 1, 1.33, 1.66 and 2. but up to you.